taichi/ir/statements.cpp

```diff
@@ -88,6 +88,18 @@ bool GlobalPtrStmt::is_element_wise(SNode *snode) const {
   return true;
 }
 
+bool GlobalPtrStmt::covers_snode(const SNode *snode) const {
+  // Check if the addresses of this statement all over the loop cover
+  // all active indices of the snode.
+  for (auto &index : indices) {
+    if (auto loop_unique = index->cast<LoopUniqueStmt>()) {
+      if (loop_unique->covers_snode(snode))
+        return true;
+    }
+  }
+  return is_element_wise(snode);
```

Comment:
I'm not entirely sure what `covers` means when the SNode has multiple indices:
```python
for i, j in x:
    p = ti.loop_unique(x[i, j], covers=z.parent())
    y[i, p] = 0 # Is `y` covered by all pairs (i, p)? According to the code here it seems so.
``` 

I think either behavior is fine - or we can limit all inputs to `covers` to be 1D fields.

Reply:
Yes, `y` is covered by all pairs (i, p) according to the code here.

I think we can limit all inputs to `covers` to be 1D fields.